model/host/hostutil.go

```diff
@@ -15,7 +16,7 @@ import (
 )
 
 func (h *Host) SetupCommand() string {
-	cmd := fmt.Sprintf("%s host setup", filepath.Join("~", h.Distro.BinaryName()))
+	cmd := fmt.Sprintf("mv '%s' '%s' && %s host setup", evergreen.TempSetupScriptName, evergreen.SetupScriptName, filepath.Join("~", h.Distro.BinaryName()))
```

Comment:
Could this be part of the setup command instead of shelling out to mv? Also, does failing the agent deploy job if the setup script is not present actually solve this problem. It still could be the case that a host gets marked provisioned and then is marked decommissioned if two agent deploy jobs run.

Reply:
I moved the rename from here to `evergreen host setup`.